### DISCUSSION TITLE
How I can set for only user who's have admin role can access the menu.

### DISCUSSION THREAD

**Discussion by jigujigu** [filamentphp/filament#6274] [Help] [answered]
Hi All,
How I can set for only user who's have admin role can access the menu.
Below code show error : **Call to undefined function App\Providers\hasRole()**

```
use Illuminate\Support\Facades\Auth;

use Filament\Facades\Filament;
use Filament\Navigation\UserMenuItem;
------------------ 
        if (Auth::user()-hasRole('admin')) {
            Filament::serving(function () {
                Filament::registerUserMenuItems([
                    UserMenuItem::make()
                        ->label('Settings')
                        ->url(route('filament.pages.settings'))
                        ->icon('heroicon-s-cog'),
                    
                ]);
            });
        }
```

**Comment by jigujigu** [comment]
change to if (auth->user()-hasRole('admin'))   give me error : Undefined constant "App\Providers\auth"

**Reply by danharrin** [reply, marked as answer]
`auth()->user()->hasRole(...)`. Make sure the `if` statement is inside the `serving()` function instead of around it.

**Reply by jigujigu** [reply]
It's worked.
But, why vscode mark hasRole  as an error ?

![image](https://user-images.githubusercontent.com/6214964/233593655-0708a369-8383-4b19-8150-578e4f77bfef.png)

Thank You

**Reply by danharrin** [reply]
This isnt a Filament question any more. Does the method exist on the user model? If you are using a package, make sure you set it up properly

**Reply by RetroCIB** [reply]
try this: 
```
$user = auth()->user();   
if($user->hasRole('admin')){
//...
}
```
